fossunited/foss_hackathon/doctype/foss_hackathon_localhost/test_foss_hackathon_localhost.py

```diff
@@ -0,0 +1,169 @@
+        self.localhost.append(
+            "organizers",
+            {
+                "profile": self.organizer1_profile.name,
+            },
+        )
+        self.localhost.save()
+
+        # Organizer 1 should have the "Localhost Organizer" Role
+        self.assertTrue(self.has_organizer_role(self.ORGANIZER_1))
+        # When self.localhost is trashed
+        self.localhost.delete(force=True)
+        # Then Organizer 1 should not have the "Localhost Organizer" Role
+        self.assertFalse(self.has_organizer_role(self.ORGANIZER_1))
+
+    def test_handle_assignment_deassignment_together(self):
+        """
+        Test that roles are handled correctly when same
+        number of organizers are added and removed simultaneously
+        """
+        # Add Organizer 1 to self.localhost
+        self.localhost.append(
+            "organizers",
+            {
+                "profile": self.organizer1_profile.name,
+            },
+        )
+        self.localhost.save()
+
+        self.assertTrue(self.has_organizer_role(self.ORGANIZER_1))
+
+        # Remove organizer 1 from self.localhost
+        for member in self.localhost.organizers:
+            if member.profile == self.organizer1_profile.name:
+                self.localhost.organizers.remove(member)
+                break
+        # Before save, add Organizer 2 to self.localhost
+        self.localhost.append(
+            "organizers",
+            {
+                "profile": self.organizer2_profile.name,
+            },
+        )
+        self.localhost.save()
+        self.assertFalse(self.has_organizer_role(self.ORGANIZER_1))
+        self.assertTrue(self.has_organizer_role(self.ORGANIZER_2))
+
+    def test_other_localhost_member(self):
+        # Add Organizer 1 to self.localhost
+        self.localhost.append(
+            "organizers",
+            {
+                "profile": self.organizer1_profile.name,
+            },
+        )
+        self.localhost.save()
+
+        self.assertTrue(self.has_organizer_role(self.ORGANIZER_1))
+
+        # Create another localhost with Organizer 1 as an organizer
+        new_localhost = insert_test_hackathon_localhost(
+            parent_hackathon=self.hackathon.name,
+            organizers=[
+                {
+                    "profile": self.organizer1_profile.name,
+                },
+            ],
+        )
+
+        # Remove Organizer 1 from self.localhost
+        for member in self.localhost.organizers:
+            if member.profile == self.organizer1_profile.name:
+                self.localhost.organizers.remove(member)
+                break
+
+        self.localhost.save()
+
+        # Organizer 1 should still have the "Localhost Organizer" Role
+        self.assertTrue(self.has_organizer_role(self.ORGANIZER_1))
+        new_localhost.delete(force=True)
+
+    def has_organizer_role(self, user):
+        roles = frappe.get_roles(user)
+        if "Localhost Organizer" in roles:
```

Comment:
One more reason why it is good to add the `Localhost Organizer` string as a constant somewhere. Not sure what other Roles are added via code similar to this